old_timesheet_migration/timesheet_migration.py

```diff
@@ -0,0 +1,186 @@
+                    time_entries = time_entries.append({'member_acronym': member_acronym,
+                                                        'project_name': project_name,
+                                                        'activity_name': activity_name,
+                                                        'client_name': client_name,
+                                                        'start': start, 'end': end}, ignore_index=True)
+    return time_entries
+
+def get_entity_id(entity, where, update=None):
+    return globals().get(entity).update_or_create(where, update).id
+
+def fill_ids(time_entries):
+    '''Receives a data frame with all time entries without ids
+        Returns a data frame with ids.'''
+
+    member_ids = {}
+    members_acronym = time_entries.member_acronym.unique()
+    for acronym in members_acronym:
+        email = {'email': acronym + "@certi.org.br"}
+        member_ids[acronym] = get_entity_id('Member', {'acronym': acronym}, email)
+
+    project_ids = {}
+    project_names = time_entries.project_name.unique()
+    for name in project_names:
+        project_ids[name] = get_entity_id('Project', {'name': name})
+
+    activity_ids = {}
+    activity_names = time_entries.activity_name.unique()
+    for name in activity_names:
+        activity_ids[name] = get_entity_id('Activity', {'name': name})
+    
+    client_ids = {}
+    client_names = time_entries.client_name.unique()
+    for name in client_names:
+        client_ids[name] = get_entity_id('Client', {'name': name})
+
+    time_entries_with_ids = update_time_entries_ids(time_entries, member_ids,
+                                                    project_ids, activity_ids, client_ids)
+    return time_entries_with_ids
+
+
+def update_time_entries_ids(time_entries, member_ids, project_ids, activity_ids, client_ids):
+    '''Receives a dataframe that represents the time_entries without the ids
+        Returns with the correct ids, based on the dictionaries member_ids,
+        project_ids, activity_ids and client_ids'''
+
+    for index, row in time_entries.iterrows():
+        time_entries.loc[index, 'member_id'] = member_ids[row[0]]
+        time_entries.loc[index, 'project_id'] = project_ids[row[2]]
+        time_entries.loc[index, 'activity_id'] = activity_ids[row[4]]
+        time_entries.loc[index, 'client_id'] = client_ids[row[6]]
+    return time_entries
+
+def send_to_database(time_entries):
+    time_entries = time_entries.reset_index()
+    for index in range(len(time_entries)):
+        TimeEntry.update_or_create({
+            "member_id": time_entries.loc[index, 'member_id'],
+            "project_id": time_entries.loc[index, 'project_id'],
+            "activity_id": time_entries.loc[index, 'activity_id'],
+            "client_id": time_entries.loc[index, 'client_id'],
+            "start": time_entries.loc[index, 'start'],
+            "end": time_entries.loc[index, 'end']})
```

Comment:
The function is missing doc string and I think you can replace this `for` for the [apply method](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.apply.html)

Reply:
@yvesfracari did you resolve this?